tests/datasets/test__init__.py

```diff
@@ -0,0 +1,424 @@
+# Copyright 2018-2019 Faculty Science Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pytest
+
+from faculty import datasets
+
+
+@pytest.fixture
+def mock_client(mocker):
+    session = mocker.Mock()
+    get_session_mock = mocker.patch(
+        "faculty.datasets.get_session", return_value=session
+    )
+
+    project_id = mocker.Mock()
+
+    object_client = mocker.Mock()
+    mocker.patch("faculty.datasets.ObjectClient", return_value=object_client)
+
+    yield object_client, project_id
+
+    get_session_mock.assert_called_once_with()
+
+
+def test_datasets_ls_all_files(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    list_response = mocker.Mock()
+    mock_object = mocker.Mock()
+    mock_object.path = "test-path"
+    list_response.objects = [mock_object]
+    list_response.next_page_token = None
+    object_client.list.return_value = list_response
+
+    objects = datasets.ls("test-prefix", project_id, show_hidden=True)
+    assert objects == ["test-path"]
+    object_client.list.assert_called_once_with(project_id, "test-prefix")
+
+
+def test_datasets_ls_non_hidden_files(mocker, mock_client):
+    object_client, project_id = mock_client
+
+    list_response = mocker.Mock()
+    mock_object = mocker.Mock()
+    mock_object.path = ".test-hidden-path"
+    list_response.objects = [mock_object]
```

Comment:
Might be good to also check that normal files are not filtered out. In this situation, the following more concise way of building `Mock` objects might be useful:

```python
list_response.objects = [mocker.Mock(path=".test-hidden"), mocker.Mock(path="not-hidden")]
```